jemalloc-sys/src/lib.rs

```diff
@@ -68,13 +497,222 @@ extern "C" {
+ 	                            arena_ind: c_uint) -> *mut c_void;
+
+/// Extent deallocation function.
+///
+/// Deallocates an extent at given `addr` and `size` with `committed`/decommited
+/// memory as indicated, on behalf of arena `arena_ind`, returning `false` upon
+/// success.
+///
+/// If the function returns `true`, this indicates opt-out from deallocation;
+/// the virtual memory mapping associated with the extent remains mapped, in the
+/// same commit state, and available for future use, in which case it will be
+/// automatically retained for later reuse.
+pub type extent_dalloc_t = fn (extent_hooks: *mut extent_hooks_t,
+                               addr: *mut c_void,
+                               size: size_t,
+                               committed: c_bool,
+                               arena_ind: c_uint) -> c_bool;
+
+/// Extent destruction function.
+///
+/// Unconditionally destroys an extent at given `addr` and `size` with
+/// `committed`/decommited memory as indicated, on behalf of arena `arena_ind`.
+///
+/// This function may be called to destroy retained extents during arena
+/// destruction (see `arena.<i>.destroy`).
+pub type extent_destroy_t = fn (extent_hooks: *mut extent_hooks_t,
+                                addr: *mut c_void,
+                                size: size_t,
+                                committed: c_bool,
+                                arena_ind: c_uint);
+
+/// Extent commit function.
+///
+/// Commits zeroed physical memory to back pages within an extent at given
+/// `addr` and `size` at `offset` bytes, extending for `length` on behalf of
+/// arena `arena_ind`, returning `false` upon success.
+///
+/// Committed memory may be committed in absolute terms as on a system that does
+/// not overcommit, or in implicit terms as on a system that overcommits and
+/// satisfies physical memory needs on demand via soft page faults. If the
+/// function returns `true`, this indicates insufficient physical memory to
+/// satisfy the request.
+pub type extent_commit_t = fn (extent_hooks: *mut extent_hooks_t,
+                               addr: *mut c_void,
+                               size: size_t,
+                               offset: size_t,
+                               length: size_t,
+                               arena_ind: c_uint) -> c_bool;
+
+/// Extent decommit function.
+///
+/// Decommits any physical memory that is backing pages within an extent at
+/// given `addr` and `size` at `offset` bytes, extending for `length` on behalf of arena
+/// `arena_ind`, returning `false` upon success, in which case the pages will be
+/// committed via the extent commit function before being reused.
+///
+/// If the function returns `true`, this indicates opt-out from decommit; the
+/// memory remains committed and available for future use, in which case it will
+/// be automatically retained for later reuse.
+pub type extent_decommit_t = fn (extent_hooks: *mut extent_hooks_t,
```

Comment:
Should these be `extern` fns?

Reply:
Yes, nice catch. I wonder why `ctest` did not catch these.